File: Leetcode/Medium/1143-Longest_Common_Subsequence.py
def longestCommonSubsequence(self, s1, s2):
    length1 = len(s1)
    length2 = len(s2)
    temp  = [0]*(1+length1)
    ls = []
    for _ in range(length2+1):
        ls.append(temp.copy())
    
    for i in range(1,length1+1):
        for j in range(1,length2+1):
            if s1[i-1] == s2[j-1]:
                ls[j][i] = 1+ls[j-1][i-1]
            else: ls[j][i] = max(ls[j][i-1],ls[j-1][i])
    
    return ls[-1][-1]



def isSubsequence(s, t):# if the first ware to accure more than once there is a way that this will fail but that's is not the case this time
    i  = 0
    slength = len(s)
    for c in t:
        if i == slength: return True
        if i<slength:
            if c == s[i]:
                i+=1
    return i==slength

File: Leetcode/Medium/1143 Longest Common Subsequence .cpp
int Max(int &a, int &b)
{
	if (a > b)
		return a;
	return b;
}
int longestCommonSubsequence(string &a, string &b)
{
	int n = a.size();
	int m = b.size();

	int dp[n + 1][m + 1];
	memset(dp, 0, sizeof(dp));

	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < m; j++)
		{
			if (a[i] == b[j])
				dp[i + 1][j + 1] = dp[i][j] + 1;
			else
				dp[i + 1][j + 1] = Max(dp[i + 1][j], dp[i][j + 1]);
		}
	}
	return dp[n][m];
}